ex03.exs

```diff
@@ -58,14 +58,82 @@ defmodule Ex03 do
         5 elegant use of language features or libraries
 
   """
+  
```

Comment:
## -10 

This is a lot of very nice looking code. However, it really shouldn't have been :)

I want you to think of coding as a series of transformations. Here, we have a collection, which we transform into a list of subcollections. We then transform this into a list of running subprocesses. We then transform the results returned from these into a single list.

On way of coding this is simply:

~~~ elixir
  defp into_chunks(collection, count) do
    chunk_size = div(Enum.count( collection), count )
    Enum.chunk_every( collection, chunk_size)
  end

  def pmap(collection, process_count, function) do
    chunk_mapper = &( Task.async( fn -> Enum.map( &1, function) end ) )

    collection
    |> into_chunks(process_count)
    |> Enum.map( chunk_mapper )
    |> Enum.map( &( Task.await( &1 ) ) )
    |> Enum.concat()
  end
~~~